homeassistant/components/homeassistant_connect_zbt2/update.py

```diff
@@ -153,6 +153,9 @@ async def async_setup_entry(
 
     async_add_entities([entity])
 
+    # Fetch firmware info early to avoid prolonged Unknown state
+    await entity.coordinator.async_request_refresh()
```

Comment:
Instead of having to add this to all hardware integrations individually, are there any downsides to adding it to the `BaseFirmwareUpdateEntity`'s `async_added_to_hass` method here? [homeassistant/components/homeassistant_hardware/update.py#L128-L151](https://github.com/home-assistant/core/blob/8dfa0f2f656388f762e1928fe20ecfe4ff1431f8/homeassistant/components/homeassistant_hardware/update.py#L128-L151)

Seems like that should work as well, kind of similar to how hassio is doing this here: [homeassistant/components/hassio/entity.py#L65](https://github.com/home-assistant/core/blob/8dfa0f2f656388f762e1928fe20ecfe4ff1431f8/homeassistant/components/hassio/entity.py#L65)

Reply:
> Is there an existing pattern in Core for coordinators that should refresh immediately upon being added? `async_added_to_hass` as both you and @TheJulianJES pointed out?

A common pattern is to call `coordinator.async_config_entry_first_refresh()` on the integration's `async_setup_entry`, but that would not work here since the coordinator is not created there. 

@TheJulianJES sorry, I've misread your suggestion and was thinking that you wanted to add it to the base `UpdateEntity`. 

Yeah, I think @TheJulianJES suggestion does make sense! Sorry for the noise!